### PR TITLE
add user to project 

### PR DESCRIPTION
**Pull request by Bogdan-Demianchuk**

### CODE REVIEW

File: src/main/java/com/cinema/service/impl/AuthenticationServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    }
+
+    @Override
+    public User register(String email, String password) {
+        User user = new User(email, password);
+        byte[] salt = HashUtil.getSalt();
+        user.setPassword(HashUtil.hashPassword(user.getPassword(), salt));
+        user.setSalt(salt);
+        userService.add(user);
+        return null;
```
Comment (boroda4436) [inline src/main/java/com/cinema/service/impl/AuthenticationServiceImpl.java:34]: чому?

File: src/main/java/com/cinema/util/HashUtil.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        StringBuilder hashedPassword = new StringBuilder();
+        try {
+            MessageDigest messageDigest = MessageDigest.getInstance(SHA);
+            messageDigest.update(salt);
+            byte[] digest = messageDigest.digest(password.getBytes());
+            for (byte b : digest) {
+                hashedPassword.append(String.format("%02x", b));
+            }
+        } catch (java.security.NoSuchAlgorithmException e) {
+            logger.error("Cant't hash a password ", e);
```
Comment (boroda4436) [inline src/main/java/com/cinema/util/HashUtil.java:28]: чому логи?

File: src/main/java/com/cinema/dao/impl/CinemaHallDaoImpl.java [hunk 3]
```diff
@@ -39,12 +38,10 @@ public CinemaHall add(CinemaHall cinemaHall) {
     @Override
     public List<CinemaHall> getAll() {
         try (Session session = HibernateUtil.getSessionFactory().openSession()) {
-            CriteriaQuery<CinemaHall> criteriaQuery = session
-                    .getCriteriaBuilder().createQuery(CinemaHall.class);
-            criteriaQuery.from(CinemaHall.class);
-            return session.createQuery(criteriaQuery).getResultList();
+            Query query = session.createQuery("from CinemaHall");
```
[the remark this answers is not in the document]
Comment (Bogdan-Demianchuk) [reply src/main/java/com/cinema/dao/impl/CinemaHallDaoImpl.java:41]: Done

File: src/main/java/com/cinema/dao/impl/CinemaHallDaoImpl.java [hunk 4]
```diff
@@ -39,12 +38,10 @@ public CinemaHall add(CinemaHall cinemaHall) {
...
+            return (List<CinemaHall>) query.list();
```
[the remark this answers is not in the document]
Comment (Bogdan-Demianchuk) [reply src/main/java/com/cinema/dao/impl/CinemaHallDaoImpl.java:42]: Done

File: src/main/java/com/cinema/dao/impl/MovieSessionDaoImpl.java [hunk 5]
```diff
@@ -40,47 +38,27 @@ public MovieSession add(MovieSession movieSession) {
...
     public List<MovieSession> findAvailableSessions(Long movieId, LocalDate date) {
-        Transaction transaction = null;
-        Session session = null;
-        try {
-            session = HibernateUtil.getSessionFactory().openSession();
-            transaction = session.beginTransaction();
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
             Query query = session
                     .createQuery("from MovieSession where movie.id = :id "
                             + "AND showTime > :time");
```
[the remark this answers is not in the document]
Comment (Bogdan-Demianchuk) [reply src/main/java/com/cinema/dao/impl/MovieSessionDaoImpl.java:55]: misunderstood available, fixed

File: src/main/java/com/cinema/dao/impl/MovieSessionDaoImpl.java [hunk 6]
```diff
@@ -40,47 +38,27 @@ public MovieSession add(MovieSession movieSession) {
...
             query.setParameter("id", movieId);
             query.setParameter("time", date.atStartOfDay());
-            List list = query.list();
-            transaction.commit();
-            return list;
+            return (List<MovieSession>) query.list();
```
[the remark this answers is not in the document]
Comment (Bogdan-Demianchuk) [reply src/main/java/com/cinema/dao/impl/MovieSessionDaoImpl.java:58]: Done

File: src/main/java/com/cinema/service/impl/AuthenticationServiceImpl.java (hunk 1 above)
Comment (Bogdan-Demianchuk) [reply src/main/java/com/cinema/service/impl/AuthenticationServiceImpl.java:34]: bug came out, fixed

File: src/main/java/com/cinema/util/HashUtil.java (hunk 2 above)
Comment (Bogdan-Demianchuk) [reply src/main/java/com/cinema/util/HashUtil.java:28]: Added Runtime

**Pull request merged**